allennlp/data/semparse/worlds/nlvr_world.py

```diff
@@ -0,0 +1,436 @@
+            if comparison_op(attribute_function(entity), target_attribute):
+                returned_set.append(entity)
+        return set(returned_set)
+
+    ## Box filtering functions
+    @classmethod
+    def filter_equals(cls,
+                      set_to_filter: Set[Box],
+                      attribute_function: Callable[[Box], AttributeType],
+                      target_attribute: AttributeType) -> Set[Box]:
+        return cls._filter(set_to_filter, attribute_function, target_attribute, operator.eq)
+
+    @classmethod
+    def filter_not_equal(cls,
+                         set_to_filter: Set[Box],
+                         attribute_function: Callable[[Box], AttributeType],
+                         target_attribute: AttributeType) -> Set[Box]:
+        return cls._filter(set_to_filter, attribute_function, target_attribute, operator.ne)
+
+    @classmethod
+    def filter_greater_equal(cls,
+                             set_to_filter: Set[Box],
+                             attribute_function: Callable[[Box], AttributeType],
+                             target_attribute: AttributeType) -> Set[Box]:
+        return cls._filter(set_to_filter, attribute_function, target_attribute, operator.ge)
+
+    @classmethod
+    def filter_lesser_equal(cls,
+                            set_to_filter: Set[Box],
+                            attribute_function: Callable[[Box], AttributeType],
+                            target_attribute: AttributeType) -> Set[Box]:
+        return cls._filter(set_to_filter, attribute_function, target_attribute, operator.le)
+
+    ## Object filtering functions
+    @classmethod
+    def black(cls, objects_set: Set[Object]) -> Set[Object]:
+        return cls._filter(objects_set, cls._get_single_color, 'black', operator.eq)
+
+    @classmethod
+    def blue(cls, objects_set: Set[Object]) -> Set[Object]:
+        return cls._filter(objects_set, cls._get_single_color, 'blue', operator.eq)
+
+    @classmethod
+    def yellow(cls, objects_set: Set[Object]) -> Set[Object]:
+        return cls._filter(objects_set, cls._get_single_color, 'yellow', operator.eq)
+
+    @classmethod
+    def circle(cls, objects_set: Set[Object]) -> Set[Object]:
+        return cls._filter(objects_set, cls._get_single_shape, 'circle', operator.eq)
+
+    @classmethod
+    def square(cls, objects_set: Set[Object]) -> Set[Object]:
+        return cls._filter(objects_set, cls._get_single_shape, 'square', operator.eq)
+
+    @classmethod
+    def triangle(cls, objects_set: Set[Object]) -> Set[Object]:
+        return cls._filter(objects_set, cls._get_single_shape, 'triangle', operator.eq)
+
+    @classmethod
+    def same_color(cls, objects_set: Set[Object]) -> Set[Object]:
+        """
+        Returns the set of objects whose color is the most frequent color in the initial set, if the highest
+        frequency is greater than 1. If not, all objects are of different colors, and this method returns a
+        null set.
```

Comment:
"returns an empty set", not "returns a null set".

Reply:
Done.